### PR TITLE
Add GraalVM JDK 21.0.2

### PR DESCRIPTION
**Pull request by moonfruit**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask-versions/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/graalvm-jdk21.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  artifact "graalvm-jdk-#{version}+13.1", target: "/Library/Java/JavaVirtualMachines/graalvm-#{version.major}.jdk"
+
+  # No zap stanza required
+
+  caveats do
+    license "https://www.oracle.com/downloads/licenses/graal-free-license.html"
+    <<~EOS
+      According to https://www.graalvm.org/jdk21/docs/getting-started/macos/
+        This app will not work with quarantine attributes.
```
Comment (bevanjkay) [inline Casks/graalvm-jdk21.rb:27]: This is the same case as the main `graalvm-jdk` cask, and the caveat is not present there.

**Comment by bevanjkay** [comment]
The token should be `graalvm-jdk-21`

**Comment by moonfruit** [comment]
JDK related casks are named like this, for example: `oracle-jdk17`, `graalvm-jdk17`, `temurin17`, `temurin21`, `zulu17`, `zulu21`.

File: Casks/graalvm-jdk21.rb (hunk 1 above)
Comment (moonfruit) [reply Casks/graalvm-jdk21.rb:27]: This part is copied from `graalvm-jdk17`, and it really can't work with `quarantine` as the main `graalvm-jdk`. I even think the main `graalvm-jdk` should also include this note.

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**